sklearn/preprocessing/_function_transformer.py

```diff
@@ -40,18 +41,45 @@ class FunctionTransformer(BaseEstimator, TransformerMixin):
         kwargs forwarded. If inverse_func is None, then inverse_func
         will be the identity function.
 
-    validate : bool, optional default=True
+    validate : bool or 'array-or-frame', optional default=True
         Indicate that the input X array should be checked before calling
-        func. If validate is false, there will be no input validation.
-        If it is true, then X will be converted to a 2-dimensional NumPy
-        array or sparse matrix. If this conversion is not possible or X
-        contains NaN or infinity, an exception is raised.
+        func. The possibilities are:
+
+        - If True, then X will be converted to a 2-dimensional NumPy array or
+          sparse matrix. If the conversion is not possible an exception is
+          raised.
+        - If False, then there is no input validation
+        - If 'array-or-frame', X will be pass-through if this is a pandas
+          DataFrame or converted to a 2-dimensional array or sparse matrix. In
+          this latest case, an exception will be raised if the conversion
+          failed.
+
+        .. deprecated:: 0.20
+           ``validate=True`` as default will be replaced by
+           ``validate='array-or-frame'`` in 0.22.
+
+        .. versionadded:: 0.20
+           ``validate`` takes the option ``'array-or-frame'``.
 
     accept_sparse : boolean, optional
         Indicate that func accepts a sparse matrix as input. If validate is
         False, this has no effect. Otherwise, if accept_sparse is false,
         sparse matrix inputs will cause an exception to be raised.
 
+    force_all_finite : boolean or 'allow-nan', optional default=True
+        Whether to raise an error on np.inf and np.nan in X. The possibilities
+        are:
+
+        - If True, force all values of X to be finite.
+        - If False, accept both np.inf and np.nan in X.
+        - If 'allow-nan', accept only np.nan values in X. Values cannot be
+          infinite.
+
+        Applied only when ``validate=True``.
```

Comment:
As I understand, in the future the default for `validate` will be `'array-or-frame'`. That means that this `force_all_finite` will not be used anymore once that change is made? 
That seems a bit strange as a) that will be a change in behaviour and b) then this default in the documentation will be misleading.

Reply:
> That means that this force_all_finite will not be used anymore once that change is made

The aim is to keep DataFrame as DataFrame with converting it (calling `check_array`). So with `'array-or-frame'` this behavior is enforce but I still keep checking for the finiteness in array or Frame.

But the documentation is wrong :)